cli/lib/tasks/process-js-task.js

```diff
@@ -0,0 +1,381 @@
+		const loaded = await this.loadTaskData();
+		const fullBuild = !loaded || this.requiresFullBuild();
+		if (fullBuild) {
+			return this.doFullTaskRun();
+		}
+
+		Object.keys(this.data.jsFiles).forEach(relPath => this.builder.unmarkBuildDirFile(this.data.jsFiles[relPath].dest));
+	}
+
+	/**
+	 * Function that will be called after the task run finished.
+	 *
+	 * Used to update the task data before it gets written to disk.
+	 */
+	async afterTaskAction() {
+		await super.afterTaskAction();
+
+		this.data.jsBootstrapFiles = this.jsBootstrapFiles;
+		this.data.jsFiles = this.jsFiles;
+		this.data.analyzeOptionsHash = this.generateHash(JSON.stringify(this.defaultAnalyzeOptions));
+		await this.saveTaskData();
+
+		this.builder.unmarkBuildDirFiles(this.incrementalDirectory);
+	}
+
+	/**
+	 * Creates the hooks that are required during JS processing.
+	 */
+	createHooks() {
+		let compileJsFileHook = this.builder.cli.createHook('build.ios.compileJsFile', this.builder, (r, from, to, done) => {
+			// Read the possibly modified file contents
+			const source = r.contents;
+
+			// If the file didn't change from previous run, return early
+			const currentHash = this.generateHash(source);
+			const previousHash = this.data.contentHashes[from];
+			if (previousHash && previousHash === currentHash) {
+				this.builder.unmarkBuildDirFile(to);
+				return done();
+			}
+
+			this.transformAndCopy(source, from, to).then(() => {
+				this.data.contentHashes[from] = currentHash;
+				return done();
+			}).catch(e => {
+				// if we have a nicely formatted pointer to syntax error from babel, print it!
+				if (e.codeFrame) {
+					this.logger.error(e.codeFrame);
+				}
+
+				done(e);
+			});
+		});
+
+		this.copyResourceHook = promisify(this.builder.cli.createHook('build.ios.copyResource', this.builder, (from, to, done) => {
+			const originalContents = fs.readFileSync(from).toString();
+
+			const r = {
+				original: originalContents,
+				contents: originalContents,
+				symbols: []
+			};
+
+			compileJsFileHook(r, from, to, done);
+		}));
+	}
+
+	/**
+	 * Processes a single JavaScript file by applying several transforms and then copying it to
+	 * its destination path.
+	 *
+	 * If the content of the unprocessed JavaScript file did not change compared to the last run,
+	 * any further processing will be skipped.
+	 *
+	 * @param {String} filePathAndName Full path to the JavaScript file
+	 * @return {Promise}
+	 */
+	async processJsFile(filePathAndName) {
+		let file = this.resolveRelativePath(filePathAndName);
+		if (!file) {
+			throw new Error(`Unable to resolve relative path for ${filePathAndName}.`);
+		}
+
+		const bootstrapPath = file.substr(0, file.length - 3);  // Remove the ".js" extension.
+		if (bootstrapPath.endsWith('.bootstrap') && !this.jsBootstrapFiles.includes(bootstrapPath)) {
+			this.jsBootstrapFiles.push(bootstrapPath);
+		}
+
+		const info = this.jsFiles[file];
+		if (this.builder.encryptJS) {
+			if (this.platform === 'ios') {
```

Comment:
May need to check/normalize from 'iphone' as well. I think the CLI still accepts that (and not sure it normalizes it to 'ios' itself).

Reply:
The CLI will normalize this to `ios`. Test with `ti build -p ios|iphone|ipad` and all resolved to `ios`.